sdk/storage/azure-storage-blobs/test/ut/blob_container_client_test.cpp

```diff
@@ -1248,6 +1274,42 @@ namespace Azure { namespace Storage { namespace Test {
       auto blobItem = GetBlobItem(blobName);
       EXPECT_EQ(blobItem.Name, blobName);
     }
+    {
+      const std::string blobPrefix
+          = std::string("aaaaa\xEF\xBF\xBF") + "bbb/"; // UTF-8 0xEF, 0xBF, 0xBF is UTF-16 0xFFFF
```

Comment:
UTF-16 0xffff is [Unicode NOT A CHARACTER](https://unicode-table.com/en/FFFF/).

That is not supposed to ever be encoded in unicode text, is that intentional? If so, you might want to explain why you're using a character that should never be encoded.

Reply:
`0xfffe` and `0xffff` are valid code points, but not valid in XML. This feature is to add a way to list blobs with these unicode characters in blob names.

Reply:
Then document this please. Every reference I saw for 0xffff and 0xfffe indicated that they weren't supposed to appear in character strings, so this raised red flags.

Reply:
added